### PR TITLE
Add OpenAI tests

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue
Part of fixing #4757 

## What does this PR do?
- OpenAI embedder: don't pass apiKey when it is empty (slightly improves error messages)
- rest embedder and rest-based embedders: specialize the authorization denied error message depending on the configuration source
- fix existing tests
- Adds assets containing prerecorded texts to embed and the embeddings obtained from OpenAI
- Adds an asset containing a tokenized long document and the embedding obtained from OpenAI for this token
- Uses the wiremock crate to mock the OpenAI API: parse the openai request, lookup the response in assets, craft an openai response

### CODE REVIEW

File: meilisearch/tests/vector/openai.rs [hunk 1]
```diff
@@ -0,0 +1,1873 @@
...
+        // json file that was compressed with gzip
+        // decompress with `gzip --keep -d openai_tokenized_responses.json.gz`
+        // recompress with `gzip --keep -c openai_tokenized_responses.json > openai_tokenized_responses.json.gz`
+        let compressed_responses = include_bytes!("openai_tokenized_responses.json.gz");
+        let mut responses = Vec::new();
+        let mut decoder = flate2::write::GzDecoder::new(&mut responses);
+
+        decoder.write_all(compressed_responses).unwrap();
+        drop(decoder);
+        serde_json::from_slice(&responses).unwrap()
```
Comment (irevoire) [inline meilisearch/tests/vector/openai.rs:118]: We may consider putting an on-disk cache on this part if it starts taking a lot of time

File: meilisearch/tests/vector/openai.rs [hunk 2]
```diff
@@ -0,0 +1,1873 @@
...
+    ]
+    "###);
+}
+
+// test with a server that responds 500 on 3 out of 4 calls
+#[actix_rt::test]
+async fn it_still_works() {
+    let (_mock, setting) = create_fallible_mock().await;
+    let server = get_server_vector().await;
+    let index = server.index("doggo");
```
Comment (irevoire) [inline meilisearch/tests/vector/openai.rs:1648]: That's a very nice test tbh
It’s awesome you did it

**Review by irevoire — APPROVED** [review]
Hey, thanks a lot that’s really some awesome tests

bors merge

**Comment by dureuill** [comment]
bors merge

**Pull request merged**